docs/docs/reference/02-yaml-syntax.md

```diff
@@ -264,471 +605,299 @@ for available event syntax.
+The `all` operator takes a list of items and is true only when every listed item is true.
 
 ```yaml
-transition: <component_name>
+condition_1:
+  all:
+    - { component: component_a, predicate: some_predicate }
+    - { component: component_b, predicate: some_predicate }
+  events:
+    ...
 ```
 
-#### Trigger a lifecycle transition
+#### Any
+
+The `any` operator takes a list of items and is true when at least one of the listed items is true.
 
 ```yaml
-lifecycle: "configure"
+condition_1:
+  any:
+    - { component: component_a, predicate: some_predicate }
+    - { component: component_b, predicate: some_predicate }
+  events:
+    ...
 ```
 
-Request a lifecycle transition on the component that is triggering the event, using one of the available transitions
-(`configure`, `activate`, `deactivate`, `cleanup`, or `shutdown`).
+#### One of
+
+The `one_of` operator takes a list of items and is true only when exactly one of the listed items is true.
 
 ```yaml
-lifecycle: activate
+condition_1:
+  one_of:
+    - { component: component_a, predicate: some_predicate }
+    - { component: component_b, predicate: some_predicate }
+  events:
+    ...
 ```
 
-Request a lifecycle transition on a different component.
+### Nested conditions
+
+The conditional operators can be applied recursively for more complex conditions. The following example could be
+collapsed into the equivalent logical pseudocode: `NOT(a AND b AND (c OR d OR (e XOR f)))`
 
 ```yaml
-lifecycle:
-  transition: activate
-  component: <component_name>
+conditions:
+  nested_condition:
+    not:
+      all:
+        - { component: component_1, predicate: a }
+        - { component: component_2, predicate: b }
+        - any:
+            - { component: component_3, predicate: c }
+            - { component: component_4, predicate: d }
+            - one_of:
+                - { component: component_5, predicate: e }
+                - { component: component_6, predicate: f }
 ```
 
-Use a list to trigger multiple transitions from a single predicate.
+## Sequences
 
-```yaml
-lifecycle:
-  - transition: activate
-    component: <component_name>
-  - transition: deactivate
-    component: <component_name>
-```
+A sequence is a list of steps that are handled sequentially in order. Sequence steps are either
+standard [state events](#events) or conditional blocks; the conditional steps are used either to wait for a condition,
+predicate or fixed time interval, or to assert the current value of a condition or predicate.
 
-#### Set a parameter
+Similar to [conditions](#conditions), sequences are listed under a top-level field called `sequences`. Sequence names
+must be unique, and should generally follow the `lower_camel_case` naming convention.
 
-Set a parameter on the component that is triggering the event.
+After sequences are defined in the yaml, they can be managed using [sequence state events](#manage-sequences); component
```

Comment:
Here, should the semicolon be a comma after `sequence state events`?

Reply:
ah, probably not.

Reply:
Ok, by reading again it indeed makes more sense. Then, perhaps I would personally prefer a full stop, but ok this is neither incomprehensible nor wrong, and I'm at my limit with grammar :D. Either is fine with me.

Thanks for putting all this together, it's a nice update (and sorry for the nitpicks).